### DISCUSSION TITLE
Can .wasm and .data sizes be further reduced?

### DISCUSSION THREAD

**Discussion by digitalsignalperson** [pthom/imgui_bundle#184] [Q&A] [answered]
I noticed most of the online demos, including [hello world](https://traineq.org/ImGuiBundle/emscripten/bin/demo_hello_world.html), have a ~2.6MB `.wasm` file and a ~6MB `.data` file. Plus maybe another ~300kB of `.js`. 

In contrast [this sokol imgui demo](https://floooh.github.io/sokol-html5/imgui-sapp.html) has only a 1.1MB `.wasm` and 30kB of `.js`.

Are there any current ways to reduce the binary sizes, or ideas for what code would need changed, or otherwise is this as good as it can be with the design of the bundle? Thanks! This is such an interesting framework to consider for multi-platform development.

**Comment by pthom** [comment, marked as answer]
Hi,

* About the .data file:
The 6MB is due to the fact that all demos share the same resources (with lots of example images and fonts). 
It can be reduced to 350Ko if you do not include an assets folder. Those 350Ko will consist mainly of the default font (DroidSans) that is used by hello imgui. The Sokol demo does not include a font (but uses the inline imgui font, which can lead to pixelated results, as you can see on the sokol demo).

* About the 2.6MB file:  imgui bundle includes a lot more that the sokol demo you linked (OpenCV, freetype, imgui, node editor, implot, immvision, etc, etc). It also links with SDL (which is responsible for a good 1MB, I guess). This can be reduced to 2.2M if compiling with `-DCMAKE_BUILD_TYPE=MinSizeRel`, and to 1.8M if using hello_imgui. It could be further reduced by using glfw instead of SDL, and maybe by using sokol (but this is a road I do not have time to take now).

* About the js: I don't know

I did my tests with https://github.com/pthom/imgui_bundle_template, where I removed the assets folder, and disabled the usage of ImGuiMd, which requires the RobotoFont to be present in the assets folder).


i.e, I compiled this file:

```
// Part of ImGui Bundle - MIT License - Copyright (c) 2022-2024 Pascal Thomet - https://github.com/pthom/imgui_bundle
#include "immapp/immapp.h"
#ifdef IMGUI_BUNDLE_WITH_IMPLOT
#include "implot/implot.h"
#endif
#include "imgui_md_wrapper.h"

#include <cmath>


void DemoImplot()
{
    static std::vector<double> x, y1, y2;
    if (x.empty())
    {
        double pi = 3.1415;
        for (int i = 0; i < 1000; ++i)
        {
            double x_ = pi * 4. * (double)i / 1000.;
            x.push_back(x_);
            y1.push_back(cos(x_));
            y2.push_back(sin(x_));
        }
    }

    ImGuiMd::Render("# This is the plot of _cosinus_ and *sinus*");
#ifdef IMGUI_BUNDLE_WITH_IMPLOT
    if (ImPlot::BeginPlot("Plot"))
    {
        ImPlot::PlotLine("y1", x.data(), y1.data(), (int)x.size());
        ImPlot::PlotLine("y2", x.data(), y2.data(), (int)x.size());
        ImPlot::EndPlot();
    }
#endif
}


void Gui()
{
    DemoImplot();

    ImGui::Separator();
}


int main(int , char *[])
{
    HelloImGui::SimpleRunnerParams runnnerParams;
    runnnerParams.guiFunction = Gui;
    runnnerParams.windowSize = {600, 800};

    ImmApp::AddOnsParams addOnsParams;
    addOnsParams.withImplot = true;

    ImmApp::Run(runnnerParams, addOnsParams);
    return 0;
}
```

and got

```
347K  imgui_bundle_example_integration.data
2.2M  imgui_bundle_example_integration.wasm
```

**Reply by digitalsignalperson** [reply]
Cool thanks for the info! That's seems pretty good
